Update About the Docs page content

laurapacilio opened the pull request:
**Goals:**
- Make the content more concise and easier to read.
- Help new users understand where they should go next. When you don't know the products or what they're used for, it can be very difficult to build a mental model of how all of the elements in the docs dropdown fit together or how to find the right docs for your use case. The new content tries to align docs content with what users are trying to do (or what they may want to do) with Terraform.
- Reduce the need for scrolling. In the previous version, you had to scroll down to get to some of the descriptions about what elements are or why you'd click them. The goal is to make **the section titles** of this intro material fit above the fold in users' laptop and desktop screens. That way, they're more likely to find what they're looking for.


I can verify that I:
- [x] Built the content locally
- [x] Assigned at least one reviewer
- [x] Added at least one label
- [x] Checked for broken links

Here's what the first draft of this looks like rendered, for context:

<img width="950" alt="Screen Shot 2021-08-12 at 11 03 28 AM" src="https://user-images.githubusercontent.com/83350965/129220510-b7066935-699e-4b2a-b00d-57ba9881b520.png">

judithpatudith commented on content/source/docs/index.html.markdown:
```diff
@@ -10,70 +10,35 @@ show_notification: false
+### Manage Infrastructure
 
-### [Learn Terraform ➜](https://learn.hashicorp.com/terraform?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) (external site)
+- Use the [Terraform Configuration Language](/docs/language/index.html) to describe infrastructure.
```
```suggestion
- Describe infrastructure in [Terraform Configuration Language](/docs/language/index.html).
```

laurapacilio replied:
good call! I pushed a change that is very close to both of these with a minor change - see what you think!

judithpatudith commented on content/source/docs/index.html.markdown:
```diff
@@ -10,70 +10,35 @@ show_notification: false
 
-### [Learn Terraform ➜](https://learn.hashicorp.com/terraform?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) (external site)
+- Use the [Terraform Configuration Language](/docs/language/index.html) to describe infrastructure.
+- Use the [Terraform CLI](/docs/cli/index.html) with the Terraform binary for command-line workflows.
```
```suggestion
- Use the [Terraform Command Line Interface (CLI)](/docs/cli/index.html) to manage configuration, plugins, infrastructure, and state.
```

judithpatudith replied:
I think there's a world where we separate docs for the CLI and the binary itself, but we aren't there yet, and you can use many of the CLI commands with Terraform Cloud too :)

judithpatudith commented on content/source/docs/index.html.markdown:
```diff
@@ -10,70 +10,35 @@ show_notification: false
 
-Documentation about developing Terraform providers, with extensive information about Terraform's internals.
+- [Terraform Cloud](/docs/cloud/index.html) helps teams use Terraform together, enabling version control, data sharing, and more.
+- [Terraform Enterprise](/docs/enterprise/index.html) offers a private instance of Terraform Cloud, with features like access control and governance.
```
```suggestion
- [Terraform Enterprise](/docs/enterprise/index.html) is a self-hosted instance of Terraform Cloud.
```

Paid tiers of TFC also include these features

judithpatudith commented on content/source/docs/index.html.markdown:
```diff
@@ -10,70 +10,35 @@ show_notification: false
+- [Terraform Enterprise](/docs/enterprise/index.html) offers a private instance of Terraform Cloud, with features like access control and governance.
 
-Terraform relies on provider plugins to manage infrastructure resources across a wide variety of infrastructure services. Anyone can make and distribute a Terraform provider for their own service.
+### Develop Plugins
```
Don't forget that you can also publish modules, which aren't plugins, on the Terraform Registry. I'm trying to think of how to retitle this section to keep it clean and also accurate, but drawing a bit of a blank... Maybe:

```suggestion
### Develop Modules and Plugins
```

judithpatudith commented on content/source/docs/index.html.markdown:
```diff
@@ -10,70 +10,35 @@ show_notification: false
+### Develop Plugins
 
+- [Create a provider](/docs/extend/index.html) to allow Terraform to interact with a service.
+- [Publish your provider](/docs/registry/index.html) to the Terraform Registry to make it publicly available.
```
```suggestion
- [Publish a provider or module](/docs/registry/index.html) to the Terraform Registry to make it publicly available.
```

judithpatudith commented on content/source/docs/index.html.markdown:
```diff
@@ -10,70 +10,35 @@ show_notification: false
+### Collaborate
 
-Documentation about developing Terraform providers, with extensive information about Terraform's internals.
+- [Terraform Cloud](/docs/cloud/index.html) helps teams use Terraform together, enabling version control, data sharing, and more.
```
```suggestion
- [Terraform Cloud](/docs/cloud/index.html) helps teams use Terraform together, with version control, state sharing, and more.
```

judithpatudith reviewed (approved):
I absolutely love this change and think its a huge navigation improvement. Added a couple wording nits and tried to respect your character count as much as possible :) 

@im2nguyen would you gut check this for me too? It's a pretty high traffic page so I think this PR could use more eyes :)

im2nguyen reviewed (approved):
This looks great! I like how you organized the sections to fit the practitioner journey 😄 (why, try, buy, fly)

I think the page looks slightly strange since there's no content before the first header... but really like how you can see everything without scrolling

laurapacilio commented:
@judithpatudith and @im2nguyen Could you take one more look? I incorporated your edits and then made a few other tiny changes. I agreed with Ty (this is why it's good to let things sit!) that it looked a bit bizarre without anything before the section titles, so I added a little description of what Terraform is at the top. I thought this would also be beneficial in case a user lands here from a google search and bypasses the product homepage completely. Thoughts?

I also added another bullet under Develop and Share (and changed the title to that). Idk - what do you think? Is it okay? too cheesy? I think it works and is nice because it's shorter than having Plugin and Module in the title. Tu I went back and forth about contribute (and almost did it!) but then I was worried that "contribute" might feel like contributing to the codebase/submitting PRs, since this is an open-source project. That was the rationale,  but still very open to other thoughts on this section title. 

Finally I added "governance" to cloud with a link to sentinel! :-)  it doesn't all quite fit above the fold anymore but I still think it's reasonable and organized and reduces scrolling quite a bit.

judithpatudith commented on content/source/docs/index.html.markdown:
```diff
@@ -3,77 +3,44 @@ layout: 'docs-frontpage'
+### Manage Infrastructure
 
-### [Learn Terraform ➜](https://learn.hashicorp.com/terraform?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) (external site)
+- Describe infrastructure with [Terraform's configuration language](/docs/language/index.html) on various [providers](/docs/language/providers/index.html).
```
```suggestion
- Describe infrastructure on various [providers](/docs/language/providers/index.html) with [Terraform's configuration language](/docs/language/index.html).
```

judithpatudith reviewed (approved):
Looks great! One last teeny wording tweak, but that's it!
